lib/galaxy/files/sources/posix.py

```diff
@@ -72,6 +76,9 @@ def _list(self, path="/", recursive=True, user_context=None, opts: Optional[File
     def _realize_to(
         self, source_path: str, native_path: str, user_context=None, opts: Optional[FilesSourceOptions] = None
     ):
+        if not self.root and (not user_context or not user_context.is_admin):
+            raise exceptions.ItemAccessibilityException(f"Non admins cannot realize files when configured to handle only file:// protocol")
```

Comment:
As a user I don't know what that is telling me, but I should probably not be informed of configuration stuff.
```suggestion
            raise exceptions.ItemAccessibilityException("Writing to file:// URLs is not allowed")
```
?

Reply:
This is a good point. I originally phrased it that way because file:// url downloads from filesources that have a root url defined would still be allowed (I'll add a test for that), provided any defined security checks pass. I've rephrased it to say disabled, let me know whether that works better.